outlines/structure.md

```diff
@@ -0,0 +1,56 @@
+# Application structure and code style
+
+1. What's different between Meteor and other kinds of apps?
+    1. All JavaScript, enables code sharing
+    2. Structure is flexible
+    3. Currently everything is about magic file names
+    4. App scope vs. package scope vs. file scope variables
+2. Client vs. server vs. both
+3. Smooth path from prototype to production, three sizes of apps, S, M, L
+4. General stuff
+    1. Filenames - named the same as the export
+        1. Every file exports one thing
+    2. Avoid depending on load order via namespacing and Meteor.startup
+        1. If you depend on load order, your best bet is a package
+    3. Naming guidelines for everything
+        1. https://github.com/meteor/guide/issues/9#issuecomment-148800085
```

Comment:
Should this really be in core for now? This is really then one big can of worms. How you then make internationalization of messages, for example? What we did for example in the past was that we added extra stuff to `Meteor.Error` on the server side so that `check` could tell which field failed, so that you could color that correctly in the form, when validation failed. But we abandoned that with a pattern that validation should not be part of method interface. Method is more like business logic API. You get it right or you get 400 error. `check` assures that. And then you build on top of this a higher-level form field component which does all the fancy stuff of validating on the client, animation error messages, and flashing red colors, and once it let's it go through, you make a method call and you expect it to succeed. And higher-level stuff can also automatically generate such methods which prepend `check`s at the beginning.

BTW, there is also the issue that it would be great that all form submissions would get disabled when connection to the server is lost. But this should probably not be part of core.